htlcswitch/hop/payload.go

```diff
@@ -373,7 +380,19 @@ func validateBlindedRouteTypes(parsedTypes tlv.TypeMap,
 	case updateAddBlindingSet && onionBlindingSet:
 		return nil, ErrInvalidPayload{
 			Type:      record.BlindingPointOnionType,
-			Violation: IncludedViolation,
+			Violation: OverloadedViolation,
+			FinalHop:  false,
+		}
+
+	// Do not accept HTLCs for which we have no blinding point in
+	// either the onion TLV payload or the UpdateAddHTLC message.
+	// The sender is trying to use route blinding, but we didn't
+	// receive the blinding point needed to decrypt the onion.
+	// The sender or the previous peer is buggy or malicious. (eclair)
```

Comment:
Note that this is when encrypted data is populated (that's how we know we're attempting to use route blinding / hit this function call). 

Looking at this now, isn't the error type here that we included encrypted data (rather than omitted one of the blinding points, since we actually don't know which one is missing).

Reply:
Don't we know exactly which or when both are missing? In this case, the blinding point is not set in either the top level onion TLV or the `UpdateAddHTLC` message. I agree tho that the way this error is structured is slightly funky given that it could just have easily used `lnwire.BlindingPointRecordType` as that blinding point is also missing.

Reply:
Given that ambiguity and the fact that we can only express a single type as included/omitted, maybe it does make sense to swap to highlighting the inclusion of the encrypted data as the source of the error. 

AFAIK, no non-test code acts on the contents of this error - whether it is one type included/missing or the other, so it may not matter much so long as the end behavior is we send `InvalidOnionBlinding` error to the upstream peer.  I'll swap over if you think it more sensible/descriptive?
```go
		return nil, ErrInvalidPayload{
			Type:      record.BlindingPointOnionType,
			Violation: OmittedViolation,
                }

		return nil, ErrInvalidPayload{
			Type:      record.EncryptedDataOnionType,
			Violation: IncludedViolation, 
                         // included when both blinding points omitted is an error
                }
```